src/encodings.jl

```diff
@@ -1,271 +1,384 @@
+    "IBM935", "IBM937", "IBM939", "IBM943", "IBM9448",
+    "iso_8859_1", "iso_8859_13", "iso_8859_15", "iso_8859_2",
+    "iso_8859_3", "iso_8859_4", "iso_8859_5", "iso_8859_6", "iso_8859_7",
+    "iso_8859_8", "iso_8859_8-i", "iso_8859_9", "iso_8859-1", "ISO_8859-1",
+    "ISO_8859-1:1987", "ISO_8859-10", "ISO_8859-10:1992", "ISO_8859-11",
+    "iso_8859-13", "ISO_8859-13", "ISO_8859-14", "ISO_8859-14:1998",
+    "iso_8859-15", "ISO_8859-15", "ISO_8859-15:1998", "ISO_8859-16",
+    "ISO_8859-16:2001", "iso_8859-2", "ISO_8859-2", "ISO_8859-2:1987",
+    "iso_8859-3", "ISO_8859-3", "ISO_8859-3:1988", "iso_8859-4",
+    "ISO_8859-4", "ISO_8859-4:1988", "iso_8859-5", "ISO_8859-5",
+    "ISO_8859-5:1988", "iso_8859-6", "ISO_8859-6", "ISO_8859-6:1987",
+    "iso_8859-7", "ISO_8859-7", "ISO_8859-7:1987", "ISO_8859-7:2003",
+    "iso_8859-8", "ISO_8859-8", "iso_8859-8-i", "ISO_8859-8:1988",
+    "iso_8859-9", "ISO_8859-9", "ISO_8859-9:1989", "ISO_8859-9E",
+    "iso-8859-1", "ISO-8859-1", "ISO-8859-10",
+    "ISO-8859-11", "iso-8859-13", "ISO-8859-13", "ISO-8859-14", "iso-8859-15",
+    "ISO-8859-15", "ISO-8859-16", "iso-8859-2", "ISO-8859-2", "iso-8859-3",
+    "ISO-8859-3", "iso-8859-4", "ISO-8859-4", "iso-8859-5", "ISO-8859-5",
+    "iso-8859-6", "ISO-8859-6", "iso-8859-7", "ISO-8859-7", "iso-8859-8",
+    "ISO-8859-8", "iso-8859-8-i", "iso-8859-9", "ISO-8859-9", "ISO-8859-9E",
+    "iso8859-1", "ISO8859-1", "ISO8859-10",
+    "ISO8859-11", "iso8859-13", "ISO8859-13", "ISO8859-14", "iso8859-15",
+    "ISO8859-15", "ISO8859-16", "iso8859-2", "ISO8859-2", "iso8859-3",
+    "ISO8859-3", "iso8859-4", "ISO8859-4", "iso8859-5", "ISO8859-5",
+    "iso8859-6", "ISO8859-6", "iso8859-7", "ISO8859-7", "iso8859-8",
+    "ISO8859-8", "iso8859-8-i", "iso8859-9", "ISO8859-9", "ISO8859-9E",
+    "ISO88591", "ISO885910", "ISO885911", "ISO885913", "ISO885914",
+    "ISO885915", "ISO885916", "ISO88592", "ISO88593", "ISO88594",
+    "ISO88595", "ISO88596", "ISO88597", "ISO88598", "ISO88599", "ISO88599E",
+    "windows-1250", "WINDOWS-1250", "windows-1251", "WINDOWS-1251", "windows-1252",
+    "WINDOWS-1252", "windows-1253", "WINDOWS-1253", "windows-1254", "WINDOWS-1254",
+    "windows-1255", "WINDOWS-1255", "windows-1256", "WINDOWS-1256",
+    "windows-1257", "WINDOWS-1257", "windows-1258", "WINDOWS-1258",
+    "WINDOWS-31J", "WINDOWS-50221", "WINDOWS-51932", "windows-874",
+    "WINDOWS-874", "WINDOWS-932", "WINDOWS-936"]
+
+# 16-bit codeunit encodings
+const encodings16 = [
+    "UTF-16", "UTF-16BE", "UTF-16LE"
+    ]
+
+# 32-bit codeunit encodings
+const encodings32 = [
+    "UTF-32", "UTF-32BE", "UTF-32LE", "UTF32", "UTF32BE", "UTF32LE"
+    ]
+
+const encodings_other = [
+    "1026", "1046", "1047", "10646-1:1993", "10646-1:1993/UCS4",
+    "437", "500", "500V1", "850", "851", "852", "855", "856", "857",
+    "860", "861", "862", "863", "864", "865", "866", "866NAV", "869",
+    "874", "8859_1", "8859_2", "8859_3", "8859_4", "8859_5", "8859_6",
```

Comment:
Yes, as I noted, there's a lot of classification work to do here. I've just started moving a few of these to `encodings8` to test how it works.

Anyway, if we want to store more properties about each encoding, we should create an immutable with a few fields, and make an array of that, instead of storing only the name.